homeassistant/components/scsgate.py

```diff
@@ -0,0 +1,154 @@
+                msg = "Exception while processing event: {}".format(
+                    exception)
+                self._logger.error(msg)
+        else:
+            self._logger.info(
+                "Ignoring state message for device {} because unknonw".format(
+                    message.entity))
+
+    @property
+    def devices(self):
+        """ Dictionary with known devices. Key is device ID,
+        value is the device itself """
+        return self._devices
+
+    def add_device(self, device):
+        """ Adds the specified device to the list of the already registered ones.
+
+        Beware: this is not what you usually want to do, take
+        a look at `add_devices_to_register`
+        """
+        self._devices[device.scs_id] = device
+
+    def add_devices_to_register(self, devices):
+        """ List of devices to be registered.
+
+        Arguments:
+        * devices: list of devices to register
+        """
+        with self._devices_to_register_lock:
+            for device in devices:
+                self._devices_to_register[device.scs_id] = device
+        self._activate_next_device()
+
+    def _activate_next_device(self):
+        """ Starts the activation of the 1st device inside of self._devices """
+        with self._devices_to_register_lock:
+            if len(self._devices_to_register) == 0:
+                return
+            _, device = self._devices_to_register.popitem()
+            self._devices[device.scs_id] = device
+            self._device_being_registered = device.scs_id
+            self._reactor.append_task(GetStatusTask(target=device.scs_id))
+
+    def is_device_registered(self, device_id):
+        """ Checks whether a device is already registered or not
+
+        Arguments:
+        device_id: the ID of the device to look for
+        """
+        with self._devices_to_register_lock:
+            if device_id in self._devices_to_register.keys():
+                return False
+
+        with self._device_being_registered_lock:
+            if device_id == self._device_being_registered:
+                return False
+
+        return True
+
+    def start(self):
+        """ Start the scsgate.Reactor """
+        self._reactor.start()
+
+    def stop(self):
+        """ Stop the scsgate.Reactor """
+        self._reactor.stop()
+
+    def append_task(self, task):
+        """ Registers a new task to be executed """
+        self._reactor.append_task(task)
+
+
+def setup(hass, config):
+    """ Setup the SCSGate component. """
+
+    _LOGGER.info(config)
+
+    def stop_monitor(event):
+        """ Invoked when home-assistant is exiting. Performs the necessary
+        cleanups """
+        _LOGGER.info("Stopping SCSGate monitor thread")
+        SCSGATE.stop()
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, stop_monitor)
+
+    device = config['scsgate']['device']
+    global SCSGATE
+    SCSGATE = SCSGate(device=device, logger=_LOGGER)
+    SCSGATE.start()
```

Comment:
Would you be able to detect if starting has failed if for example hardware is missing?